Assets/Scripts/LoadingScreen.cs

```diff
@@ -0,0 +1,69 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class LoadingScreen : MonoBehaviour
+{
+    public float loadTime = 10;
+    public Slider ProgressBar;
+    public List<LoadedAsset> Assets;
+    public List<LoadedAsset> ResourcesAssets;
+    public List<LoadedAsset> StreamingAssets;
+    private CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+    private CancellationToken cancellationToken;
+
+    private async void Start()
+    {
+        cancellationToken = cancellationTokenSource.Token;
+        await LoadAllAssets();
+        await LoadAllAssetsFromResources();
+        await LoadAllAssetsFromStreamingAssets();
+    }
+
+    public async Task LoadAllAssets()
+    {
+        foreach (var asset in Assets)
```

Comment:
you are not awaiting in parallel, what will happen is that you will first wait for the first item, then the second, and so forth.
read about Task.WhenAll